apis/services.d.ts

```diff
@@ -171,7 +171,7 @@ export class Service extends QueryAPI {
     <T = any>(details: { event: types.event; data?: object; headers?: object }): Promise<T>
     <T = any>(details: { query: ConstructedQuery; data?: object; headers?: object }): Promise<T>
     <T = any>(details: { method: types.eventName; path: string; data?: object; headers?: object }): Promise<T>
-    <T = any>(details: { event: types.eventName; entity: LinkedDefinition | string; data?: object; params?: object }): Promise<T>
+    <T = any>(details: { event: types.eventName; entity: LinkedDefinition | string; data?: object; params?: object; headers?: object }): Promise<T>
```

Comment:
`headers` appears to be a recurring parameter type throughout the services definition. If I gather this from CAPire correctly, we are referring to [general HTTP headers](https://nodejs.org/api/http.html#http_message_headers) here, is that correct? If so, we could probably use [`import type { IncomingHttpHeaders} from 'node:http'`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/http.d.ts#L49) type across all signatures. That would give us a nice, strict typing, consistency (currently we have a mix of `object` and `{}) and code completion:

![Screenshot 2024-01-08 at 07 45 01](https://github.com/cap-js/cds-types/assets/103028279/0010105b-b40a-4fb6-9bce-21b4ae597f64)

Reply:
For most use cases, I think that would be an options. However, the messages of headers are of a different format.
Here, we'd need to have additional type, I suppose.